freeipa/src/main/java/com/sequenceiq/freeipa/service/user/UsersyncService.java

```diff
@@ -56,6 +57,34 @@ public void synchronizeUsers(String accountId, SynchronizeUsersRequest request)
         LOGGER.info("Done!");
     }
 
+    public void setPassword(String accountId, String username, String password) {
+        LOGGER.info("UsersyncService.synchronizeUsers() called");
+
+        List<Stack> stacks = stackService.getAllByAccountId(accountId);
+
+        if (stacks.isEmpty()) {
+            LOGGER.warn("No stacks found for accountId {}", accountId);
+            return;
+        }
+
+        // TODO propagate exceptions out through API
+        for (Stack stack : stacks) {
+            FreeIpaClient freeIpaClient;
+            try {
+                freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStack(stack);
+            } catch (Exception e) {
+                LOGGER.error("Failed to create FreeIpaClient for stack {}", stack.getId(), e);
+                continue;
```

Comment:
if we move the next try-catch into this one, then this continue could be omitted

Reply:
work and exception handling moved to SetPasswordHandler.